builder.go

```diff
@@ -174,7 +176,7 @@ func (b *Builder) Build() (*BuildMetadata, error) {
 
 	return &BuildMetadata{
 		BOM:        bom,
-		Buildpacks: b.Group.Group,
```

Comment:
1. Instead of replacing the group the `group.toml` with an enhanced group here I think we actually want to add `homepage` to `group.toml` -https://github.com/buildpacks/spec/pull/127/files

2. The merged spec change doesn't actually add `homepage` to `metadata.toml`. I'll grant that may be unexpected in that this is the first time `metadata.toml` and the `io.buildpacks.build.metadata` label are diverging. However, the label currently contains a lot of information we don't need/want inside the contianer filesystem (e.g. the bom) and I expect these to diverge further in the future (I have a RFC planned on this topic).